src/sentry/snuba/metrics/extraction.py

```diff
@@ -620,13 +611,37 @@ def apdex_tag_spec(project: Project, argument: Optional[str]) -> List[TagSpec]:
     ]
 
 
-# This is used to map a metric to a function which generates a specification
-_DERIVED_METRICS: Dict[MetricOperationType, TagsSpecsGenerator] = {
-    "on_demand_failure_count": failure_tag_spec,
-    "on_demand_failure_rate": failure_tag_spec,
-    "on_demand_apdex": apdex_tag_spec,
+DERIVED_METRICS = {
+    "apdex": {
+        "tag_spec_func": apdex_tag_spec,
+        "hash_method": "op_plus_arg",
+    },
+    "failure_count": {
+        "tag_spec_func": failure_tag_spec,
+        "hash_method": "op",
+    },
+    "failure_rate": {
+        "tag_spec_func": failure_tag_spec,
+        "hash_method": "op",
+    },
 }
 
+# This is used to map a metric to a function which generates a specification
+_DERIVED_METRICS: Dict[MetricOperationType, TagsSpecsGenerator] = {}
+for metric in DERIVED_METRICS.keys():
```

Comment:
`.keys()` is almost never needed in python (unless you are doing setops on a dictionary's parts) -- iterating a dictionary iterates its keys already